spec.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: page
+title: Semantic Specification
+---
+
+## Concepts and Terminology
+
+A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system.
+
+A **Span** represents a logical unit of work in the system that has a start time and an end time. Spans may be nested and ordered to model parent-child and casual relationships. A Trace can be thought of as a tree of Spans.
+
+Every Span has zero or more **Logs**, each of which being a timestamped message with an optional structured data payload of arbitrary size.
+
+Every Span may also have zero or more key/value **Tags**, which do not have timestamps and simply annotate the spans.
+
+A **Trace Context** encapsulates the smallest amount of state needed to describe a Span's identity within a larger, potentially distributed trace, sufficient to propagate the context of a particular trace between processes.
+
+**Trace Attributes** are key/value pairs stored in a Trace Context and propagated _in-band_ to all future child Spans. Given a full-stack OpenTracing integration, Trace Attributes enable powerful functionality by transparently propagating arbitrary application data: e.g., from a mobile app all the way into the depths of a storage system. Trace attributes come with powerful _costs_ as well; since the attributes are propagated in-band, if they are too large they can reduce system throughput or contribute to RPC latency. Use Trace Attributes with care.
+
+**Trace Attributes** vs. **Span Tags**
+
+* Trace Attributes are propagated in-band across process boundaries. Span Tags are not propagated.
+* Span Tags are recorded off-band in the tracing system's storage. Trace Attributes are not recorded.
```

Comment:
Trace attributes could be recorded. Ex. in zipkin, the propagated ids are recorded. Also, the DEBUG propagation flag is recorded into the span (so that it can be processed by an out-of-band collection)